clang/unittests/Format/FormatTest.cpp

```diff
@@ -28188,6 +28188,17 @@ TEST_F(FormatTest, BreakBinaryOperations) {
                "                  | byte_buffer[2] << 16\n"
                "                  | byte_buffer[3] << 24;",
                Style);
+
+  Style.BreakBinaryOperations = FormatStyle::BBO_OnePerLine;
+  // Check operator >> special case
+  verifyFormat("std::cin\n"
+               "    >> longOperand1\n"
+               "    >> longOperand2\n"
+               "    >> longOperand3\n"
+               "    >> longOperand4\n"
+               "    >> longOperand5\n"
+               "    >> longOperand6;",
```

Comment:
```suggestion
  // Check operator>> special case.
  verifyFormat("std::cin\n"
               "    >> longOperand_1\n"
               "    >> longOperand_2\n"
               "    >> longOperand_3_;",
```
to test the case where the statement length just exceeds the column limit.

Reply:
You can add the same minimal test case at line 28095.

Reply:
I have added several more tests one per test case that are in the function:
- BBO_Never
- BBO_OnePerLine
- BBO_RespectPrecedence
- BBO_OnePerLine + BOS_NonAssignment
- BBO_RespectPrecedence + BOS_NonAssignment